portality/api/v1/discovery.py

```diff
@@ -65,15 +65,14 @@ def allowed(query, wildcards=False, fuzzy=False):
 
     return True
 
-
 class DiscoveryApi(Api):
 
     @classmethod
-    def _make_query(cls, q, page, page_size, subs):
+    def _sanitise(cls, q, page, page_size, sort, search_subs, sort_subs):
         if not allowed(q):
```

Comment:
can you make q from this point on a deepcopy of the q you get passed in? The DAO was doing this same thing and it ended up modifying the q object that's passed in, resulting in very subtle bugs. Well, not being able to delete articles is not so subtle I guess.

One way of doing that: I've modified the DAO to name the parameter passed in ```theq``` (star trek anyone?) and then the first line of the method is to make q = deepcopy(theq).